plugins/csharp/parser/src/csharpparser.cpp

```diff
@@ -0,0 +1,168 @@
+#include <csharpparser/csharpparser.h>
+
+#include <boost/filesystem.hpp>
+
+#include <model/buildaction.h>
+#include <model/buildaction-odb.hxx>
+#include <model/buildsourcetarget.h>
+#include <model/buildsourcetarget-odb.hxx>
+#include <model/file.h>
+#include <model/file-odb.hxx>
+
+#include <parser/sourcemanager.h>
+
+#include <util/logutil.h>
+#include <util/odbtransaction.h>
+#include <util/threadpool.h>
+
+#include <memory>
+
+namespace cc
+{
+namespace parser
+{
+
+CsharpParser::CsharpParser(ParserContext& ctx_): AbstractParser(ctx_)
+{
+  _threadNum = _ctx.options["jobs"].as<int>();
+}
+
+bool CsharpParser::acceptProjectBuildPath(const std::vector<std::string>& path_)
+{
+  return path_.size() >= 2 && fs::is_directory(path_[0]) && fs::is_directory(path_[1]);
+}
+
+bool CsharpParser::parse()
+{        
+  bool success = true;
+
+  std::vector<std::string> paths = _ctx.options["input"].as<std::vector<std::string>>();
+  
+    if (acceptProjectBuildPath(paths))
+    {
+      LOG(debug) << "C# parser parse path: " << paths[0];
+      LOG(debug) << "Parsed csharp project build path: " << paths[1];
+      success = success && parseProjectBuildPath(paths);
+    }
+    else
+    {
+      LOG(error) << "Build path must be a directory!";
+      success = false;
+    }
+  
+  return success;
+}
+
+bool CsharpParser::parseProjectBuildPath(const std::vector<std::string>& paths_)
+{
+  namespace ch = std::chrono;
+  fs::path csharp_path = fs::system_complete("../lib/csharp/");
+
+  std::future<std::string> log;
+
+  std::string command("./CSharpParser ");
+  command.append("'");
+  command.append(_ctx.options["database"].as<std::string>());
+  command.append("' '");
+  command.append(paths_[0]);
+  command.append("' '");
+  command.append(paths_[1]);
+  command.append("' '");
+  command.append(csharp_path.string());
+  command.append("' ");
+  command.append(std::to_string(_ctx.options["jobs"].as<int>()));
+  LOG(debug) << "CSharpParser command: " << command;
+
+  ch::steady_clock::time_point begin = ch::steady_clock::now();
+  
+  int result = bp::system(command, bp::start_dir(csharp_path), bp::std_out > log);
```

Comment:
CodeCompass parser will fail in case of calling it using a relative path. Not finding the C# binary, due to the relative path might cause this issue. The following usage will cause a failure:
 `~/CodeCompass_install/bin/CodeCompass_parser -f -w ~/[...] -n unit2 -i ~/[...] -i ~/[...] -d "[...]"`
This error message will be shown:
`termiante called after throwing an instance of 'boost::process::process_error
what():executive failed: No such file or directory 
Aborted (core dumped)`

Reply:
Fixed by 48f9983212824cff5f9aea9bc118c25149d03b71.